### PR TITLE
SST Developer Documentation

### PR DESCRIPTION
**Pull request by jleidel**
Includes:
- Developing extensions to SST
- Developing Custom SST components
- Developing SST Subcomponents
- Connecting Components with Ports
- Integration Testing, Integration with PyBind

Examples are tested/verified against SST 11.1.0 release and SST MASTER branch (04/12/2022 @ 16:21 CST)

### CODE REVIEW

File: Tutorials/Level3Tutorial.md [hunk 1]
```diff
@@ -0,0 +1,2583 @@
...
+4. Register the component with the SST Core
+5. Integrate the component into a simulation script
+6. Execute the simulation
+
+Using this development workflow, there are a number of key items to keep 
+in mind when implementing external components.
+
+* C++ Standards: The SST Core utilizes C++11 data structures and methods.  External 
+elements should follow the C++11 standard as well.
+* All external elements should implement and register a method that serves as a
```
Comment (gvoskuilen) [inline Tutorials/Level3Tutorial.md]: A clock is not necessary for all components, purely event driven components are fine.

File: Tutorials/Level3Tutorial.md [hunk 2]
```diff
@@ -0,0 +1,2583 @@
...
+
+```
+//
+// _basicComponent_h_
+//
+
+#ifndef _BASIC_COMPONENT_H_
+#define _BASIC_COMPONENT_H_
+
+#include <sst/core/sst_config.h>
```
Comment (gvoskuilen) [inline Tutorials/Level3Tutorial.md]: sst_config.h should not be included in header files (but should be included in implementation files)

File: Tutorials/Level3Tutorial.md (hunk 1 above)
Comment (jleidel) [reply Tutorials/Level3Tutorial.md]: fixed in 9feecc2

File: Tutorials/Level3Tutorial.md (hunk 2 above)
Comment (jleidel) [reply Tutorials/Level3Tutorial.md]: fixed in 9feecc2

**Comment by jleidel** [comment]
ping??

**Comment by jleidel** [comment]
I checked the `sst-info` output from the SST 12.0.1 release and it appears to match what is in the documentation.  We can update the docs once there is a new release of SST-Core

**Comment by jleidel** [comment]
closing this PR.

**Pull request closed without merging**